src/Common/DNSResolver.cpp

```diff
@@ -136,6 +136,15 @@ static DNSResolver::IPAddresses resolveIPAddressImpl(const std::string & host)
 
     DNSResolver::IPAddresses addresses = hostByName(host);
 
+    if (prefer_ipv6)
+    {
+        /// Move IPv4 addresses to the end of the container
+        [[maybe_unused]] auto first_ipv4_addr = std::remove_if(addresses.begin(), addresses.end(), [](const Poco::Net::IPAddress & element)
+        {
+           return element.family() == Poco::Net::AddressFamily::IPv4;
+        });
```

Comment:
std::partition would be a better choice here, though result is the same, but remove_if is a little bit misleading...

Reply:
Cool, I tried to find this one.